crates/testing/src/helpers.rs

```diff
@@ -393,3 +538,40 @@ pub fn build_fake_view_with_leaf_and_state(
         },
     }
 }
+
+pub async fn check_external_events<TYPES: NodeType, S: StreamExt<Item = Event<TYPES>> + Unpin>(
```

Comment:
Curious as to what this is doing that isn't handled by the testing macro?

Reply:
See remark below. This is not necessary since the network task will transform the raw event into a HotShot event, so the checks will be redundant.